### ISSUE TITLE
OpenGL ES 2.0 Compatibility

### ISSUE THREAD

**Issue by kvark** [servo/webrender#624]
We need to evaluate a GLES2 rendering path to support older mobile phones by the Quantum Renderer.
Concerned features are:
  - vertex texture fetches. We might require `GL_MAX_VERTEX_TEXTURE_IMAGE_UNITS` be something positive, and/or we'll need to combine our textures into a fewer number in order to fit.
  - instanced draws. We may need to duplicate the data per vertex and issue normal draw calls instead. This would work well with #615 for the vertex shader code would not need to be changed.
  - texture arrays. That may be a pain to emulate with multiple textures.

Basically, for each GLES3+ feature we use, we need to decide if we can require the appropriate GLES2 extension, or we have a workaround.

**Comment by jrmuizel** [comment]
As a point of reference GL_MAX_VERTEX_TEXTURE_IMAGE_UNITS is 8 on ES2 vc4 gpu in the Raspberry Pi

**Comment by glennw** [comment]
We also use dynamic for loops in some fragment shaders (e.g. the blur shaders), which is not supported in ES2. We'd need to have some way of compiling shaders with fixed blur radii on demand, or perhaps some kind of multi-pass emulation.

**Comment by est31** [comment]
Is this a dupe of #3701?

**Comment by mirh** [comment]
Sounds more like the contrary considering this has more technical info.
